runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeOreOverlay.java

```diff
@@ -65,60 +71,68 @@ public Dimension render(Graphics2D graphics)
 		int runiteFound = session.getRuniteFound();
 
 		// If no ores have even been collected, don't bother showing anything
-		if (nuggetsFound == 0 && coalFound == 0 && goldFound == 0 && mithrilFound == 0
-			&& adamantiteFound == 0 && runiteFound == 0)
+		// This will statement is true iff all ore counts are 0
+		if ((nuggetsFound | coalFound | goldFound | mithrilFound | adamantiteFound | runiteFound) == 0)
 		{
 			return null;
 		}
 
-		panelComponent.getChildren().add(TitleComponent.builder().text("Ores found").build());
-
-		if (nuggetsFound > 0)
-		{
-			panelComponent.getChildren().add(LineComponent.builder()
-				.left("Nuggets:")
-				.right(Integer.toString(nuggetsFound))
-				.build());
-		}
-
-		if (coalFound > 0)
-		{
-			panelComponent.getChildren().add(LineComponent.builder()
-				.left("Coal:")
-				.right(Integer.toString(coalFound))
-				.build());
-		}
-
-		if (goldFound > 0)
+		if (config.showLootAsItems())
 		{
-			panelComponent.getChildren().add(LineComponent.builder()
-				.left("Gold:")
-				.right(Integer.toString(goldFound))
-				.build());
+			panelComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.GOLDEN_NUGGET, nuggetsFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.COAL, coalFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.GOLD_ORE, goldFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.MITHRIL_ORE, mithrilFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.ADAMANTITE_ORE, adamantiteFound, true)));
+			panelComponent.getChildren().add(new ImageComponent(itemManager.getImage(ItemID.RUNITE_ORE, runiteFound, true)));
```

Comment:
This should only display collected ores, and not add images for ores which have 0 quantity.

Reply:
Good call, thought blast furnace showed all ores so that is why I did it this way but I missed the crucial if (count == 0) { continue; } in that overlay.